auth/types.go

```diff
@@ -68,15 +72,17 @@ func (a *Response) TokenResponse() *TokenResponse {
 	return tokenResponse
 }
 
-// Options contains information to configure Authenticate method calls
+// Options contains information to configure Authenticate method calls.
 type Options struct {
-	UserID        *string
-	ServiceClaims map[string]interface{}
-	Su            bool
-	TokenExpiry   *time.Duration
+	UserID        *string                // Optional user id
+	ServiceClaims map[string]interface{} // Optional JWT service claims
+	Su            bool                   // Indicates if token should contain the `su` claim
+	TokenExpiry   *time.Duration         // Optional token expiry (defaults to 24 hours)
 }
 
-// Payload specifies the grant type for the token
+// Payload specifies the grant type for the token.
+// Currently the only supported grant type is "client_credentials",
+// passing anything else other than this will return an error
 type Payload struct {
 	GrantType string
```

Comment:
I don't really know what other options `GrantTyoe` might be in the future, but is a string sufficiently expressive here?